frontend/cypress/integration/common/graph_toolbar.ts

```diff
@@ -0,0 +1,190 @@
+import { Before, Given, Then, When } from 'cypress-cucumber-preprocessor/steps';
+import { CytoscapeGlobalScratchData, CytoscapeGlobalScratchNamespace } from '../../../src/types/Graph';
+
+const url = '/console';
+
+Before(() => {
+  // Copied from overview.ts.  This prevents cypress from stopping on errors unrelated to the tests.
+  // There can be random failures due timeouts/loadtime/framework that throw browser errors.  This
+  // prevents a CI failure due something like a "slow".  There may be a better way to handle this.
+  cy.on('uncaught:exception', (err, runnable, promise) => {
+    // when the exception originated from an unhandled promise
+    // rejection, the promise is provided as a third argument
+    // you can turn off failing the test in this case
+    if (promise) {
+      return false;
+    }
+    // we still want to ensure there are no other unexpected
+    // errors, so we let them fail the test
+  });
+});
+
+When('user graphs {string} namespaces with refresh {string} and duration {string}', (namespaces, refresh, duration) => {
+  cy.visit(url + `/graph/namespaces?refresh=${refresh}&duration=${duration}&namespaces=${namespaces}`);
+});
+
+When('user clicks graph tour', () => {
+  cy.get('button#graph-tour').click();
+});
+
+When('user closes graph tour', () => {
+  cy.get('div[role="dialog"]').find('button[aria-label="Close"]').click();
+});
+
+When('user clicks graph traffic menu', () => {
+  cy.get('button#graph-traffic-dropdown').click();
+});
+
+When('user disables all traffic', () => {
+  cy.get('div#graph-traffic-menu').within(() => {
+    cy.get('input#grpc').should('exist').uncheck();
+    cy.get('#loading_kiali_spinner').should('not.exist');
+    cy.get('input#http').should('exist').uncheck();
+    cy.get('#loading_kiali_spinner').should('not.exist');
+    cy.get('input#tcp').should('exist').uncheck();
+    cy.get('#loading_kiali_spinner').should('not.exist');
```

Comment:
Probably a `should('not.be.disabled')` check would eliminate the need to check for the loading spinner since the toolbar checkboxes all get disabled until the graph re-render completes. 
```
cy.get('input#grpc').should('not.be.disabled').uncheck();
cy.get('input#http').should('not.be.disabled').uncheck();
```

Reply:
yes, that would probably look nicer.  next time i'm in this file I may make that change.